control-plane/catalog/to-consul/resource.go

```diff
@@ -999,3 +1025,26 @@ func (t *ServiceResource) isIngressService(key string) bool {
 func consulHealthCheckID(k8sNS string, serviceID string) string {
 	return fmt.Sprintf("%s/%s", k8sNS, serviceID)
 }
+
+// Sets the passing service weight
+func setServiceWeight(weight string, appsCount int, r *consulapi.CatalogRegistration) error {
+	// error validation if the input param is a number
+	weightI, err := strconv.Atoi(weight)
+	if err != nil {
+		return err
+	}
+
+	if weightI <= 1 {
+		return nil
+	}
+
+	var perAppWeight = weightI / appsCount
```

Comment:
I think we can set weight to whatever value we are getting.

Reply:
1. One simple reason being - if a customer is setting weight - x for a service - and if they query the will get another value from APIs which will be divided.
2. Weight are proportional - if you have two services with weights 10 and 20 respectively, the service with weight 20 would receive twice as much traffic as the service with weight 10. If we divide, we will not get the correct traffic.

Reply:
yup! we should set the weight that is set on the annotation and we dont need the app count for the reasons @absolutelightning described!